middle_end/flambda/simplify/env/rec_uses.mli

```diff
@@ -0,0 +1,83 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*    Pierre Chambart and Guillaume Bury, OCamlPro                        *)
+(*                                                                        *)
+(*   Copyright 2021--2021 OCamlPro SAS                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Rec uses
+
+    This modules aims at tracking use of mainly variables
+    (maybe other things later on ?), with the aim of:
+    - removing unused parameters of *recursive* continuations
+    - moving allocations out of the hot path of recursive
+      continuations (e.g. the allocation of a float that was unboxed
+       by simplify)
+*)
+
+type t
+(** The type tracking the uses of variables (passed through the downwards
+    accumulator.
+    This contains a stack to track in which continuation's handlers
+    the downwards acc currently is. *)
+
+val print : Format.formatter -> t -> unit
+(** Print to a formatter. *)
+
+
+(* {2 Creation and updates} *)
+
+val empty : t
+(** Empty uses *)
+
+val init_toplevel : Continuation.t -> Variable.t list -> t -> t
+(** Initialize the rec_uses so that the stack consists of a single
+    toplevel continuation. *)
+
+val stack_cont : Continuation.t -> Variable.t list -> t -> t
+(** Add a new continuation on the stack. Used when entering a
+    continuation handler. *)
+
+val unstack_cont : Continuation.t -> t -> t
+(** Pop the current top of the stack. Used when exiting the current
+    contionuation handler. *)
+
+val add_binding : Variable.t -> Name_occurrences.t -> t -> t
```

Comment:
`record_...` would probably be better names than `add_...` for those functions. I'm not certain.

Reply:
Indeed `add_` might not be the best name, @lthls any ideas/suggestions ?

Reply:
`enter_binding` ?

Reply:
it's not really correct to say `enter` here I'd say (as oppposed to say `enter_continuation`); as Pierre said, it's more of a "record that this binding is here at this point"